js/ui/widget/ui.widget.js

```diff
@@ -525,34 +525,32 @@ const Widget = DOMComponent.inherit({
     _hoverEndHandler: noop,
 
     _toggleActiveState($element, value) {
-        this._toggleHoverClass(!value);
+        this.option('isActive', value);
         $element.toggleClass('dx-state-active', value);
     },
 
-    _refreshHoveredElement(hoveredElement) {
-        const selector = this._activeStateUnit || this._eventBindingTarget();
+    _updatedHover() {
+        const hoveredElement = this._options.silent('hoveredElement');
 
-        this._forgetHoveredElement();
-        this._hoveredElement = hoveredElement.closest(selector);
-        this._toggleHoverClass(true);
+        this._hover(hoveredElement, hoveredElement);
     },
 
-    _forgetHoveredElement() {
-        this._toggleHoverClass(false);
-        delete this._hoveredElement;
-    },
+    _hover($el, $previous) {
+        const { hoverStateEnabled, disabled, isActive } = this.option();
+        const findHoverTarget = $el => $el && $el.closest(this._activeStateUnit || this._eventBindingTarget());
```

Comment:
Can we change the argument name of the `findHoverTarget` function because this name is the same with argument of the main function?

Also, can we extract the `findHoverTarget` function outside `_hover`?